src/CLI.hs

```diff
@@ -223,114 +218,32 @@ hevm actspec cid sol' solver' timeout _ = do
   let act = validation (\_ -> error "Too bad") id (enrich <$> compile specContents)
   bytecode <- fmap fromJust $ solcRuntime cid solContents
   let actbehvs = translateAct act
-  sequence_ $ flip fmap actbehvs $ \(behvs,calldata) ->
+  sequence_ $ flip fmap actbehvs $ \(name,behvs,calldata) ->
     Solvers.withSolvers solver' 1 (naturalFromInteger <$> timeout) $ \solvers -> do
```

Comment:
I think we can pull this out to the top level to avoid spawning and killing a solver on each run. We can also add some `-j` parameter or similar to control how many solvers we should spawn, and default that to the available number of cores on the system to support parallel solving.

Reply:
I've already pulled that out in a subsequent PR (#156), where I only have one top-level `withSolvers`. I can also add a `-j` option.